src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -275,7 +275,9 @@ def forward(self, x, seq_len=None):
             emb = torch.cat((freqs, freqs), dim=-1).to(x.device)
             self.cos_cached = emb.cos()[None, None, :, :]
             self.sin_cached = emb.sin()[None, None, :, :]
-        return self.cos_cached[:seq_len, ...].to(x.device), self.sin_cached[:seq_len, ...].to(x.device)
+        cos = self.cos_cached[:seq_len, ...].to(x.device).to(x.dtype)
+        sin = self.sin_cached[:seq_len, ...].to(x.device).to(x.dtype)
```

Comment:
nit: you can set dtype and device in one `to` call

```suggestion
        cos = self.cos_cached[:seq_len, ...].to(x.device, dtype=x.dtype)
        sin = self.sin_cached[:seq_len, ...].to(x.device, dtype=x.dtype)
```

Reply:
Thanks for the comment!  It's reflected to the patch!